### PR TITLE
Project Workflows & Servlet Alignment

### PR DESCRIPTION
**Pull request by thehabes**
This PR contains work done on the following issues, which align the behavior of the `/createProject`, `/copyProject`, `/copyProjectAndAnnotations`, `/project/{projectid}`, and `/project/{projectid}/manifest.json` endpoints.

#7
#37
#39 
#40 
#41

### CODE REVIEW

File: Source Packages/java/edu/slu/tpen/servlet/CreateProjectServlet.java [hunk 1]
```diff
@@ -175,7 +172,7 @@ public String creatManuscriptFolioProject(HttpServletRequest request, HttpServle
                                     JSONObject image = images.getJSONObject(n);
                                     JSONObject resource = image.getJSONObject("resource");
                                     String imageName = resource.getString("@id");
-                                    int folioKey = textdisplay.Folio.createFolioRecordFromNewBerry(city, canvas.getString("label"), imageName, archive, m.getID(), 0); //why imageName.replace('_', '&')
+                                    int folioKey = textdisplay.Folio.createFolioRecordFromNewBerry(city, canvas.getString("label"), imageName, archive, -1, 0); //why imageName.replace('_', '&')
```
Comment (cubap) [inline Source Packages/java/edu/slu/tpen/servlet/CreateProjectServlet.java:175]: I assume -1 doesn't break anything downstream that we've found. It is a good falsey value, but something like 987654 might be better.

File: Source Packages/java/edu/slu/tpen/servlet/GetProjectTPENServlet.java [hunk 2]
```diff
@@ -107,13 +106,6 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                             Folio[] folios = proj.getFolios();
                             jsonMap.put("ls_fs", gson.toJson(folios));
 //                            System.out.println("folios json ========== " + gson.toJson(folios));
-                            //get manuscripts
```
Comment (cubap) [inline Source Packages/java/edu/slu/tpen/servlet/GetProjectTPENServlet.java:110]: ha

File: Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
                     String redirectURL = req.getContextPath() + "/newberryTrans.html?projectID=" + projID;
                     resp.sendRedirect(redirectURL);
                 } else {
-                    projID = Integer.parseInt(req.getPathInfo().substring(1).replace("/", ""));
+                    projID = Integer.parseInt(req.getPathInfo().substring(1).replace("/", "").replace("manifest.json",""));
```
Comment (cubap) [inline Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java:68]: I don't think parseInt() cares about anything after the number, but this shouldn't hurt either. I mean `435/manifest.json` and `435/TiggersAreOnly1` should both work.

File: Source Packages/java/edu/slu/tpen/transfer/JsonLDExporter.java [hunk 4]
```diff
@@ -89,7 +89,7 @@ public JsonLDExporter(Project proj, User u) throws SQLException, IOException {
 
          List<Map<String, Object>> pageList = new ArrayList<>();
          for (Folio f : folios) {
-            pageList.add(buildPage(proj.getProjectID(), projName, f, u, man));
+            pageList.add(buildPage(proj.getProjectID(), f, u, man));
```
Comment (cubap) [inline Source Packages/java/edu/slu/tpen/transfer/JsonLDExporter.java:92]: Is this really called for?

**Review by cubap — APPROVED** [review]

File: Source Packages/java/edu/slu/tpen/servlet/CreateProjectServlet.java (hunk 1 above)
Comment (thehabes) [reply Source Packages/java/edu/slu/tpen/servlet/CreateProjectServlet.java:175]: Not that I have seen.  I am not attached to `-1` and agree that if you aren't attached don't use it b/c it is just one of those things.

File: Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java (hunk 3 above)
Comment (thehabes) [reply Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java:68]: Yeah, but then it isn't a pattern, it is a wildcard.  I want it to NOT work when you ask for it incorrectly.  So I disagree and think `http://paleo.rerum.io/TPEN-NL/project/4080/getFirstPage` should fail to return anything.

File: Source Packages/java/edu/slu/tpen/transfer/JsonLDExporter.java (hunk 4 above)
Comment (thehabes) [reply Source Packages/java/edu/slu/tpen/transfer/JsonLDExporter.java:92]: Hahaha I didn't do it on purpose!  I would have to change things haha.  Let it have an attitude.

File: Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java (hunk 3 above)
Comment (cubap) [reply Source Packages/java/edu/slu/tpen/servlet/ProjectServlet.java:68]: I agree. I tested this out after I commented. In Java, Integer.parseInt() needs a whole number string and nothing else, which I didn't know. In JavaScript, `parseInt("21gimli")` is `21` which is crazy, but means that if this behaved that way, `/getFirstPage` wouldn't break anything.

I like it how it is. Good work.

**Pull request merged**